Assets/Scripts/Scriptable Objects/RoamBehaviour.cs

```diff
@@ -1,25 +1,36 @@
 ﻿using UnityEngine;
+using System.Collections;
 
 [CreateAssetMenu(fileName = "Roam Behaviour", menuName = "Behaviours/Roam")]
 public class RoamBehaviour : AIBehaviour
 {
 	public override void Act(ref Knowledge knowledge)
 	{
-		// If the agent is close to it's destination, choose a new one
-		if (knowledge.Agent.remainingDistance < 1)
+		//If lure isn't active then roam normally
+		if (knowledge.LurePosition == Vector3.zero)
```

Comment:
This is an extremely unlikely edge case, but a nullable Vector3 (`Vector3?`) would work better to solve the issue of a lure actually being at `(0,0,0)`